File: src/main/java/arcanitor/civilengineering/command/BridgeCommand.java
package arcanitor.civilengineering.command;

import arcanitor.civilengineering.CivilEngineering;
import arcanitor.civilengineering.bridge.MessageHandler;
import com.google.common.collect.Lists;
import net.minecraft.command.CommandBase;
import net.minecraft.command.ICommandSender;
import net.minecraft.server.MinecraftServer;

import javax.annotation.Nonnull;
import java.util.List;

public class BridgeCommand extends CommandBase {
    private final List<String> aliases;

    public BridgeCommand(){
        aliases = Lists.newArrayList(CivilEngineering.MODID,"bridge","BRIDGE");
    }

    @Override
    @Nonnull
    public String getName() {
        return "bridge";
    }

    @Override
    public String getUsage(ICommandSender sender) {
        return "bridge <connect|disconnect>";
    }

    @Override
    @Nonnull
    public List<String> getAliases() {
        return aliases;
    }

    @Override
    public void execute(MinecraftServer server, ICommandSender sender, String[] args) /*throws CommandException*/ {
        if (args.length < 1) {
            //throw new WrongUsageException("")
            return;
        }
        String cmd = args[0];
        if (cmd.toLowerCase().equals("connect")) {
            if(!CivilEngineering.incomingMessageThread.isAlive()) {
                CivilEngineering.incomingMessageThread = new Thread(new MessageHandler());
                CivilEngineering.incomingMessageThread.start();
            }
        } else if (cmd.toLowerCase().equals("disconnect")) {
            CivilEngineering.incomingMessageThread.interrupt();
        }
    }


}

File: src/main/kotlin/civilengineering/command/BridgeCommand.kt
package civilengineering.command

import civilengineering.*
import civilengineering.CivilEngineering.logger
import civilengineering.bridge.MessageHandler
import com.google.common.collect.Lists
import net.minecraft.command.CommandBase
import net.minecraft.command.ICommandSender
import net.minecraft.server.MinecraftServer


class BridgeCommand : CommandBase() {
    private val aliases: List<String>

    init {
        aliases = Lists.newArrayList(MODID, "bridge", "BRIDGE")
    }

    override fun getName(): String {
        return "bridge"
    }

    override fun getUsage(sender: ICommandSender): String {
        return "bridge <connect|disconnect>"
    }

    override fun getAliases(): List<String> {
        return aliases
    }

    override fun execute(server: MinecraftServer, sender: ICommandSender, args: Array<String>) /*throws CommandException*/ {
        if (args.isEmpty()) {
            //throw new WrongUsageException("")
            return
        }
        //TODO: check if sender is OP or test if normal users cannot send this


        val cmd = args[0].toLowerCase()
        when (cmd) {
            "connect" -> if (MessageHandler.start()) {
                logger.info("Connected to matterbridge relay")
            } else {
                logger.error("Connection to matterbridge relay failed.")
            }
            "disconnect" -> MessageHandler.stop()
        }
    }


}
